packages/sn-controls-react/src/fieldcontrols/Number.tsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * @module FieldControls
+ */
+import TextField from '@material-ui/core/TextField'
+import Typography from '@material-ui/core/Typography'
+import React, { Component } from 'react'
+import { CurrencyFieldSetting, NumberFieldSetting } from '@sensenet/default-content-types'
+import InputAdornment from '@material-ui/core/InputAdornment'
+import { ReactClientFieldSetting } from './ClientFieldSetting'
+
+/**
+ * Interface for Number state
+ */
+export interface NumberState {
+  value: string
+}
+
+/**
+ * Field control that represents a Number field. Available values will be populated from the FieldSettings.
+ */
+export class NumberComponent extends Component<
+  ReactClientFieldSetting<NumberFieldSetting | CurrencyFieldSetting>,
+  NumberState
+> {
+  state: NumberState = {
+    value:
+      (this.props.content && this.props.content[this.props.settings.Name]) != null
+        ? this.props.content![this.props.settings.Name]
+        : this.props.settings.DefaultValue || '',
+  }
+
+  public handleChange = (e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement | HTMLSelectElement>) => {
+    this.setState({ value: e.target.value })
+    this.props.fieldOnChange && this.props.fieldOnChange(this.props.settings.Name, e.target.value)
+  }
+
+  /**
+   * Returns steps value by decimal and step settings
+   */
+  public defineStepValue = () => {
+    if (this.props.settings.Step) {
+      return this.props.settings.Step
+    }
+    if (!this.props.content) {
+      return 1
+    }
+    return Number.isInteger(this.props.content[this.props.settings.Name]) ||
+      this.props.settings.Type === 'IntegerFieldSetting'
+      ? 1
+      : 0.1
+  }
+
+  /**
+   * Returns inputadornment by currency and percentage settings
+   */
+  public defineCurrency = () => {
+    // Todo: Currency
```

Comment:
☝️

Reply:
I've tried to do that but couldn't figure out the proper way. Sensenet only provides a Format property which is a culture code. Not really helpful. Not sure how this should be implemented. Any thoughts?